homeassistant/components/sensor/__init__.py

```diff
@@ -486,3 +548,20 @@ def __repr__(self) -> str:
             return f"<Entity {self.name}>"
 
         return super().__repr__()
+
+    @callback
+    def async_registry_entry_updated(self) -> None:
+        """Run when the entity registry entry has been updated."""
+        assert self.registry_entry
+        if (
+            (sensor_options := self.registry_entry.options.get(DOMAIN))
+            and (custom_unit := sensor_options.get("unit"))
+            and self.device_class in UNIT_CONVERSIONS
+            and (device_class := self.device_class) in UNIT_CONVERSIONS
```

Comment:
Isn't this double?
```suggestion
            and (device_class := self.device_class) in UNIT_CONVERSIONS
```

Reply:
Fixed in 425ccc0dd2a955834e5f98205916bf7513ad89e2